pkg/ddc/base/clean_up.go

```diff
@@ -0,0 +1,70 @@
+package base
+
+import (
+	"fmt"
+	"time"
+
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetRemainingTimeToCleanUp return not nil remaining time if data operation has completion time and set ttlAfterFinished
+func GetRemainingTimeToCleanUp(object client.Object, opStatus *datav1alpha1.OperationStatus, operateType datav1alpha1.OperationType) (*time.Duration, error) {
+	if len(opStatus.Conditions) == 0 {
+		// data operation has no completion time
+		return nil, nil
+	}
+
+	ttl, err := GetTTL(object, operateType)
+	if err != nil {
+		return nil, err
+	}
+	if ttl == nil {
+		// if data operation does not set ttlAfterFinished, return nil
+		return nil, nil
+	}
+
+	curTime := time.Now()
+	completionTime := opStatus.Conditions[0].LastProbeTime
```

Comment:
Why the completionTIme is lastProbeTime?

Reply:
Currently, we have no `jobCompletionTime` so I use `lastProbeTime` instead. And when job complete, it is actually same as `jobCompletionTime`.

Reply:
Can we avoid using conditions[0] which is not sure  for jobCompletionTime?

Reply:
Now, data operation will be cleaned up regardless of complete or failure if ttlAfterFinished has been set. Do you mean we only clean up completed data operation?

Reply:
I mean only clean up completed or failed data operation.

Reply:
ok.